### PR TITLE
[CPU] Fix shape mismatching in fusing per channel

### PR DESCRIPTION
**Pull request by xuchen-intel**
### Details:
 - *Extend to support depthwise post ops fusing when channel_axis != 1, especially for reduce node*

### Tickets:
 - *62846*

### CODE REVIEW

**Comment by xuchen-intel** [comment]
@dmitry-gorokhov Hi Dmitry, could you please take a look?
As the infrastructures evolves a lot since my previous https://github.com/openvinotoolkit/openvino/pull/7544,  I simply make this new PR than rebasing on  https://github.com/openvinotoolkit/openvino/pull/7544. Thanks!

**Comment by dmitry-gorokhov** [comment]
@EgorDuplensky Please review the PR

**Comment by xuchen-intel** [comment]
@chenhu-wang @zhangYiIntel Could you please take a look?

File: src/plugins/intel_cpu/src/nodes/eltwise.cpp [hunk 1]
```diff
@@ -2138,40 +2138,44 @@ void Eltwise::appendPostOpsImpl(mkldnn::post_ops& ops, const VectorDims &postOpD
         default: IE_THROW() << errorPrefix << "as post operation is not supported";
         }
     } else {
-        const size_t chIdx = postOpDims.size() > 1 ? getFusingAxis() : 0;
+        size_t channelSize = 1;
+        if (static_cast<int>(channelAxis) >= 0) {
```
Comment (zhangYiIntel) [inline src/plugins/intel_cpu/src/nodes/eltwise.cpp:2142]: `size_t` cast to `int` is not good, if channelAix is size_t, it should be positive. If it has negative value, why not using int for type ?

File: src/plugins/intel_cpu/src/nodes/reduce.cpp [hunk 2]
```diff
@@ -2922,6 +2922,23 @@ bool Reduce::canApplyJIT(const Precision &input_prec, const Precision &output_pr
            std::find(std::begin(supportedPrecisions), std::end(supportedPrecisions), output_prec) != std::end(supportedPrecisions);
 }
 
+size_t Reduce::getFusingAxis() const {
+    size_t channelAxis = 1;
+    if (!keep_dims) {
+        for (auto &raw_axis : raw_axes) {
+            int axis = raw_axis >= 0 ? raw_axis : raw_axis + static_cast<int>(getInputShapeAtPort(REDUCE_DATA).getRank());
+            if (axis == 1) {
```
Comment (zhangYiIntel) [inline src/plugins/intel_cpu/src/nodes/reduce.cpp:2930]: cast `-1` to negative value is not good, why not keep it as `-1` and change `size_t` to `int` ? Is there any condition, we cannot cover ?

File: src/plugins/intel_cpu/src/nodes/eltwise.cpp (hunk 1 above)
Comment (xuchen-intel) [reply src/plugins/intel_cpu/src/nodes/eltwise.cpp]: Fixed. Thanks Yi!

File: src/plugins/intel_cpu/src/nodes/reduce.cpp (hunk 2 above)
Comment (xuchen-intel) [reply src/plugins/intel_cpu/src/nodes/reduce.cpp:2929]: Fixed. Thanks Yi!

**Review by zhangYiIntel — APPROVED** [review]
looks good to me, thx!

**Review by chenhu-wang — APPROVED** [review]

**Pull request merged**